solr/solrj/src/java/org/apache/solr/common/cloud/ClusterState.java

```diff
@@ -38,7 +43,7 @@
  * {@link ZkStateReader#getClusterState()}.
  * @lucene.experimental
  */
-public class ClusterState implements JSONWriter.Writable {
+public class ClusterState implements JSONWriter.Writable , SolrCluster {
```

Comment:
Little worried that this design allows for casting of the SolrCluster reference...

Reply:
We can have multiple implementations of `SolrCluster` .The idea of existing classes implementing these interfaces is to have readily available implementations

Reply:
Let's rename Solr luster to Cluster.